docs/content/concepts/file-operations.md

```diff
@@ -0,0 +1,374 @@
+   for the same data file.
+
+
+## Alter Table
+Execute the following statement to configure full-compaction:
+```sql
+ALTER TABLE T SET ('full-compaction.delta-commits' = '1');
+```
+
+It will create a new schema for Paimon table, namely `schema-1`, but no snapshot
+has actually used this schema yet until the next commit.
+
+This configuration will ensure that partitions are full compacted before writing
+ends, and since we haven't done any compaction yet, the next commit will produce
+two snapshots, one for data written and the other for full-compaction. We will
+not use it in this example since Flink does not support running compaction in 
+Flink SQL. However, when performing CDC ingestion with this table property, you 
+can notice two snapshots created in the next commit.
+
+{{< img src="/img/file-operations-4.png">}}
+
+
+
+
+## Physical Deletion of Files
+
+Remind that the marked data files are not truly deleted until the snapshot expires and 
+no consumer depends on the snapshot. For more information, see [Expiring Snapshots]({{< ref "maintenance/expiring-snapshots" >}}).
+
+
+
+## Flink Stream Write
+
+Finally, we will examine Flink Stream Write by utilizing the example
+of CDC ingestion. This section will address the capturing and writing of 
+change data into Paimon, as well as the mechanisms behind asynchronous compact 
+and snapshot commit and expiration.
+
+To begin, let's take a closer look at the CDC ingestion workflow and 
+the unique roles played by each component involved.
+
+{{< img src="/img/cdc-ingestion-topology.png">}}
+
+1. `MySQL CDC Source` uniformly reads snapshot and incremental data, with `SnapshotReader` reading snapshot data 
+   and `BinlogReader` reading incremental data, respectively.
+2. `Paimon Sink` writes data into Paimon table in bucket level. The `CompactManager` within it will trigger compaction
+   asynchronously.
+3. `Committer Operator` is a singleton responsible for committing and expiring snapshots.
+
+Next, we will go over end-to-end data flow. 
+
+
+{{< img src="/img/cdc-ingestion-source.png">}}
+
+`MySQL Cdc Source` read snapshot and incremental data and emit them to downstream after normalization.
+
+{{< img src="/img/cdc-ingestion-write.png">}}
+
+
+`Paimon Sink` first buffers new records in a heap-based LSM tree, and flushes them to disk when 
+the memory buffer is full. Note that each data file written is a sorted run. At this point, no manifest file and snapshot
+is created. Right before Flink checkpoint takes places, `Paimon Sink` will flush all buffered records and send committable message 
+to downstream, which is read and committed by `Committer Operator` during checkpoint.
+
+{{< img src="/img/cdc-ingestion-commit.png">}}
+
+During checkpoint, `Committer Operator` will create a new snapshot and associate it with manifest lists so that the snapshot  
+contains information about all data files in the table. `Committer Operator` will check against snapshot expiration and perform 
```

Comment:
`Committer Operator will check against snapshot expiration and perform physical deletion of marked data files.`
I think we can move this line to after compaction.

Reply:
done.